masq/src/commands/scan_command.rs

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2019, MASQ (https://masq.ai) and/or its affiliates. All rights reserved.
+
+use crate::command_context::CommandContext;
+use crate::commands::commands_common::{transaction, Command, CommandError};
+use clap::{App, Arg, SubCommand};
+use masq_lib::messages::{ScanType, UiScanRequest, UiScanResponse};
+use std::fmt::Debug;
+use std::str::FromStr;
+
+pub const SCAN_COMMAND_TIMEOUT_MILLIS: u64 = 10000;
+
+#[derive(Debug)]
+pub struct ScanCommand {
+    name: String,
+}
+
+pub fn scan_subcommand() -> App<'static, 'static> {
+    SubCommand::with_name("scan")
+        .about("Orders the Node to perform an immediate scan of the indicated type")
+        .arg(
+            Arg::with_name("name")
+                .help("Type of the scan that should be triggered")
+                .index(1)
+                .possible_values(&["payables", "receivables", "pendingpayables"])
+                .required(true)
+                .case_insensitive(true),
+        )
+}
+
+impl Command for ScanCommand {
+    fn execute(&self, context: &mut dyn CommandContext) -> Result<(), CommandError> {
+        let input = UiScanRequest {
+            scan_type: match ScanType::from_str(&self.name) {
+                Ok(st) => st,
+                Err(s) => panic!("clap schema does not restrict scan type properly: {}", s),
+            },
+        };
+        let result = transaction::<UiScanRequest, UiScanResponse>(
+            input,
+            context,
+            SCAN_COMMAND_TIMEOUT_MILLIS,
+        );
+        match result {
+            Ok(_response) => Ok(()),
+            Err(e) => Err(e),
+        }
```

Comment:
What did you have in mind?

Reply:
I may be wrong here, but I'd rather see a message saying the operation completed. Otherwise you may wonder if it did...

But... hmmm... with the messaging to UI printing logs from info higher the user will see a log because scanners logs with info at a certain point, if I remember well.